src/core/fetchers/segment/prioritized_segment_fetcher.ts

```diff
@@ -40,19 +40,23 @@ export interface ISegmentFetcherInterruptedEvent { type : "interrupted" }
 export interface IEndedTaskEvent { type : "ended" }
 
 /** Event sent by a `IPrioritizedSegmentFetcher`. */
-export type IPrioritizedSegmentFetcherEvent<T> = ISegmentFetcherEvent<T> |
-                                                 ISegmentFetcherInterruptedEvent |
-                                                 IEndedTaskEvent;
+export type IPrioritizedSegmentFetcherEvent<SegmentDataType> =
```

Comment:
I renamed the `T` part, because I lost track of what that generic part was about here.

Naming generics `T` for `Type` is originally a convention (dating from Java?) that is pretty respected in TypeScript to facilitate the distinction between real types and generic types.

However, I found than in this particular code, the generic part was refering to something unclear so I prefer to set a more indicative name.

Reply:
I'm wondering if we should add a prefix to explicitely say that this is a generic type. Something such as "GenericSegmentDataType" or event just "GSegmentDataType". And also, should we prefix types differently than an interface (T... instead of I....) ?

Reply:
Some types are just disguised interfaces, so maybe some could be prefixed T and others I.